middlewares/auth.js

```diff
@@ -21,8 +21,10 @@ passport.use(
 
 export const auth = (req, res, next) => {
   passport.authenticate("jwt", { session: false }, (error, user) => {
-    if (!user || error)
+    const { token } = req.headers?.authorization?.split(" ");
```

Comment:
To nie zadziała bo split  nie zwróci obiektu a tablice. Więc nie może być {} tylko [] i musisz też wydzielić ten barere.
const [_bearer, token] = req.headers?.authorization?.split(" ");

ewentualnie możesz też tak zapisać jeśli uważasz że jest bardziej dla CB czytelne
const token = req.headers?.authorization?.split(" ")[1];